src/main/java/eu/dissco/core/digitalmediaobjectprocessor/service/FdoRecordService.java

```diff
@@ -44,14 +61,37 @@ private JsonNode buildSingleHandleRequest(DigitalMediaObject mediaObject) {
     return request;
   }
 
-  private JsonNode generateAttributes(DigitalMediaObject mediaObject) {
+  private JsonNode generateAttributes(DigitalMediaObject mediaObject) throws PidCreationException {
     var attributes = mapper.createObjectNode();
     attributes.put(FDO_PROFILE.getAttribute(), FDO_PROFILE.getDefaultValue());
     attributes.put(DIGITAL_OBJECT_TYPE.getAttribute(), DIGITAL_OBJECT_TYPE.getDefaultValue());
     attributes.put(ISSUED_FOR_AGENT.getAttribute(), ISSUED_FOR_AGENT.getDefaultValue());
-    attributes.put(REFERENT_NAME.getAttribute(),mediaObject.type() + " for " + mediaObject.digitalSpecimenId());
-    attributes.put(MEDIA_URL.getAttribute(), getMediaUrl(mediaObject.attributes()));
-    attributes.put(SUBJECT_ID.getAttribute(), mediaObject.digitalSpecimenId());
+    attributes.put(REFERENT_NAME.getAttribute(),
+        mediaObject.type() + " for " + mediaObject.digitalSpecimenId());
+
+    attributes.put(PRIMARY_MEDIA_ID.getAttribute(), getMediaUrl(mediaObject.attributes()));
+    attributes.put(PRIMARY_MO_TYPE.getAttribute(), PRIMARY_MO_TYPE.getDefaultValue());
+    attributes.put(LINKED_DO_PID.getAttribute(), mediaObject.digitalSpecimenId());
+
+    attributes.put(IS_DERIVED_FROM_SPECIMEN.getAttribute(),
+        Boolean.valueOf(IS_DERIVED_FROM_SPECIMEN.getDefaultValue()));
+    attributes.put(LINKED_DO_TYPE.getAttribute(), LINKED_DO_TYPE.getDefaultValue());
+    attributes.put(PRIMARY_MO_ID_TYPE.getAttribute(), PRIMARY_MO_ID_TYPE.getDefaultValue());
+    attributes.put(PRIMARY_MO_ID_NAME.getAttribute(), PRIMARY_MO_ID_NAME.getDefaultValue());
+    attributes.put(RIGHTSHOLDER_PID_TYPE.getAttribute(), RIGHTSHOLDER_PID_TYPE.getDefaultValue());
+    try {
```

Comment:
Not that important, but it is often best practise to fail-fast. So these lines could fail the whole method, they might be better put first so that we don't do any unnecessary actions.